src/Workspaces/SharedUtilitiesAndExtensions/Workspace/VisualBasic/LanguageServices/VisualBasicAddImportsService.vb

```diff
@@ -38,8 +39,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.AddImports
 
         End Function
 
-        Protected Overrides Function GetGlobalImports(compilation As Compilation) As ImmutableArray(Of SyntaxNode)
-            Dim generator = VisualBasicSyntaxGenerator.Instance
+        Protected Overrides Function GetGlobalImports(compilation As Compilation, generator As SyntaxGenerator) As ImmutableArray(Of SyntaxNode)
             Dim result = ArrayBuilder(Of SyntaxNode).GetInstance()
```

Comment:
so here's what i don't understand.  Why does `VisualBasicSyntaxGenerator.Instance` not work here, but referencing `SyntaxGenerator` does work?

Reply:
SyntaxGenerator is a public type, while C# and VB implementations of it are internal types. This code is now in the shared layer, and has no access to internal types.

Reply:
Ah.  So would this work when i move those impls down though?

Reply:
Yes, that is correct. Otherwise, the code in shared layer will have to be similar to any other public consumer of Workspaces assembly. I am not sure how easy or tricky it would be move those implementations into shared layer though. I'd let you give it a shot once this goes in.